src/table/utils/handle-scroll.js

```diff
@@ -1,11 +1,30 @@
-export const handleScroll = (evt, tableContainerRef) => {
+export const handleHorizontalScroll = (evt, rtl, memoedContainer) => {
   evt.stopPropagation();
-  let { scrollLeft } = tableContainerRef.current;
-  const newScrollLeft = scrollLeft + evt.deltaX;
-  const max = tableContainerRef.current.scrollWidth - tableContainerRef.current.offsetWidth;
-  if (max > 0 && (newScrollLeft < 0 || newScrollLeft > max)) {
-    evt.preventDefault();
-    scrollLeft = Math.max(0, Math.min(max, newScrollLeft));
+  // scrollWidth is a measurement of the width of an element's content, including content not visible on the screen due to overflow.
+  // offsetWidth is a measurement in pixels of the element's CSS width, including any borders, padding, and vertical scrollbars
+  // scrollLeft is the number of pixels scrolled from its left edge
+  if (rtl) {
+    // scrollLeft is 0 when the scrollbar is at its rightmost position
+    // (at the start of the scrolled content),
+    // and then increasingly negative as you scroll towards (left) the end of the content .
+    // evt.deltaX increasingly negative as you scroll towards left,
+    // increasingly positive as you scroll towards right
+    let { scrollLeft } = memoedContainer;
+    const ScrollLeftWidth = scrollLeft + evt.deltaX;
```

Comment:
These rows are still in both the if and the else. If you move the desctructuring to the commentes on lines 3-5 it makes more sense

Reply:
good point. Updated.